### PR TITLE
Working on presenter template creation.

### PR DESCRIPTION
**Pull request by branflake2267**
Working on the presenter template creation. 

I'm moving away from the property files for options. I've refactored and moved things around. Things are still in flux, but I feel like the layout is stabilizing and I'm moving forward with the presenter creation. With the presenter, there are many options. Although the velocity template feels quite handy and is speeding the decoupling of source writing so its nice. The challenge for me is deriving all the options and proper configuration, remembering what goes where. I have a few grey areas of how things could be configured, where, when and why. Let me know if something looks out of place specifically on ui configuration selection. I figure some of the stuff will come out once I start generating the class for real. 

Directory layout so far. 
screen shot 2013-05-18 at 7 45 41 pm

Look at the many options:
screen shot 2013-05-18 at 7 11 17 pm

### CODE REVIEW

File: src/main/java/com/arcbees/plugin/template/presenter/CreateNestedPresenter.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        // TODO
+    }
+
+    private void processParentContentSlot() {
+        // TODO
+    }
+    
+    private String renderFileName(String fileName) {
+        String name = presenterOptions.getName();
+        name = name.replace(".vm", "");
```
Comment (branflake2267) [inline src/main/java/com/arcbees/plugin/template/presenter/CreateNestedPresenter.java]: Just a note .vm allows use of a velocity template editor, which is handy for editing the templates.

File: src/main/resources/com/arcbees/plugin/template/presenter/nested/__name__Presenter.java.vm [hunk 2]
```diff
@@ -1,4 +1,24 @@
-package {package};
+#if( $uihandlers )
+	#set( $implements = "implements ${name}UiHandlers" )
+	#set( $extendsView = ", HasUiHandlers<${name}UiHandlers>" )
+#else
+	#set( $implements = "" )
+	#set( $extendsView = "" )
+#end
+
```
Comment (branflake2267) [inline src/main/resources/com/arcbees/plugin/template/presenter/nested/__name__Presenter.java.vm:8]: I'm going to clean the new lines during a source formatting once the source is added through the IDE plugin.

File: src/main/java/com/arcbees/plugin/template/presenter/CreateNestedPresenter.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        processView();
+        processViewBinder();
+        
+        processNameTokens();
+        processGinModule(); // TODO ?
+        processParentContentSlot(); // TODO ?
+        // TODO ?
+    }
+
+    private void processModule() {
```
Comment (branflake2267) [inline src/main/java/com/arcbees/plugin/template/presenter/CreateNestedPresenter.java]: I've been looking at boiling these down, but haven't yet. :)

File: src/main/java/com/arcbees/plugin/template/domain/presenter/CreatedPopupPresenter.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.arcbees.plugin.template.domain.presenter;
+
+public class CreatedPopupPresenter {
+
```
Comment (christiangoudreau) [inline src/main/java/com/arcbees/plugin/template/domain/presenter/CreatedPopupPresenter.java:20]: rm 20

File: src/main/java/com/arcbees/plugin/template/domain/presenter/CreatedPresenterWidget.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.arcbees.plugin.template.domain.presenter;
+
+public class CreatedPresenterWidget {
+
```
Comment (christiangoudreau) [inline src/main/java/com/arcbees/plugin/template/domain/presenter/CreatedPresenterWidget.java:20]: rm 20

File: src/main/java/com/arcbees/plugin/template/domain/presenter/PopupPresenterOptions.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.arcbees.plugin.template.domain.presenter;
+
+public class PopupPresenterOptions {
+
```
Comment (christiangoudreau) [inline src/main/java/com/arcbees/plugin/template/domain/presenter/PopupPresenterOptions.java:20]: rm 20

**Comment by christiangoudreau** [comment]
Brandon, fit a way to hide the advanced screen. It must appear as simple as possible to the user.

Gatekeepers should not be in advanced though, it's a commonly used feature.

GOOD JOB Btw :D

**Comment by branflake2267** [comment]
Good idea. Sounds good and thanks :)

**Pull request merged**